File: Java/src/com/dbc/algorithm/ReferOffer/YongLiangGeZhanShiXianDuiLieLcof.java
package com.dbc.algorithm.ReferOffer;

import java.util.Stack;

public class YongLiangGeZhanShiXianDuiLieLcof {
    private Stack<Integer> stackFront;
    private Stack<Integer> stackBack;
    public YongLiangGeZhanShiXianDuiLieLcof() {
        stackFront = new Stack<>();
        stackBack = new Stack<>();
    }

    public void appendTail(int value) {
        while (!stackBack.isEmpty()) {
            stackFront.push(stackBack.pop());
        }
        stackFront.push(value);
    }

    public int deleteHead() {
        while (!stackFront.isEmpty()) {
            stackBack.push(stackFront.pop());
        }
        if (!stackBack.isEmpty()) {
            return stackBack.pop();
        } else {
            return -1;
        }
    }
}

File: core/tmp/Python/refer-offer/yong_liang_ge_zhan_shi_xian_dui_lie_lcof.py
class CQueue:

    def __init__(self):
        self.stack_front = list()
        self.stack_back = list()


    def appendTail(self, value: int) -> None:
        while self.stack_back:
            self.stack_front.append(self.stack_back.pop())
        
        self.stack_front.append(value)

    def deleteHead(self) -> int:
        while self.stack_front:
            self.stack_back.append(self.stack_front.pop())
        
        if self.stack_back:
            return self.stack_back.pop()
        else:
            return -1